File: examples/functions/rest-api-methods/posts/[category]/[slug]/get.go
package main

import "encoding/json"

// GET /posts/:category/:slug — both params merged into event
func handler(event map[string]interface{}) interface{} {
	category, _ := event["category"].(string)
	slug, _ := event["slug"].(string)

	body, _ := json.Marshal(map[string]interface{}{
		"category": category,
		"slug":     slug,
		"title":    category + "/" + slug,
		"url":      "/posts/" + category + "/" + slug,
	})
	return map[string]interface{}{
		"status":  200,
		"headers": map[string]string{"Content-Type": "application/json"},
		"body":    string(body),
	}
}

File: examples/functions/rest-api-methods/posts/[category]/[slug]/get.rs
use serde_json::{json, Value};

// GET /posts/:category/:slug — both params merged into event
pub fn handler(event: Value) -> Value {
    let category = event["category"].as_str().unwrap_or("");
    let slug = event["slug"].as_str().unwrap_or("");

    json!({
        "status": 200,
        "headers": { "Content-Type": "application/json" },
        "body": serde_json::to_string(&json!({
            "category": category,
            "slug": slug,
            "title": format!("{}/{}", category, slug),
            "url": format!("/posts/{}/{}", category, slug),
        })).unwrap()
    })
}
